libs/user-facing-errors/src/query_engine.rs

```diff
@@ -331,3 +331,10 @@ pub struct MissingFieldsInModel {
 pub struct ValueFitError {
     pub details: String,
 }
+
+#[derive(Debug, UserFacingError, Serialize)]
+#[user_facing(code = "P2034", message = "Assertion violation on the database: `{database_error}`")]
```

Comment:
Are there more than this one we know about?

Reply:
I have asked Garren whether we already have a user-facing code for "protocol violation" / "assertion violation" cases in which no error code or code location is received from the database, and it seems we don't have.
"Assertion violation" is a common term for when the database (or its driver) is asked to do something that breaks its preconditions (e.g., we passed some input that is too large to be processed)